### ISSUE TITLE
flutter_player: no audio when iPhone ringer is silenced

### ISSUE THREAD

**Issue by mymikemiller** [flutter/flutter#17911]
Using the example code [here](https://github.com/flutter/plugins/tree/master/packages/video_player) and running on an iPhone X, the video works fine, but the audio only works when the iPhone's ringer switch is not on "silent". 

Audio from a video player should be audible even when the ringer switch is silenced.

P.S. I would have opened this issue on the [Plugins](https://github.com/flutter/plugins) page, but there is no "Issues" tab there.

**Comment by bdoubleu86** [comment]
Looks like I am having the same / a similar issue with flutter v1.12.13+hotfix.5 and chewie player which is using video_player 0.10.5

Also this is set already:
`    // Allow audio playback when the Ring/Silent switch is set to silent
    [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback error:nil];`

With flutter v1.7.8 it still worked. Is this issue know? Can anyone help me? Which additional information do you need?

**Comment by ramiechaarani** [comment]
Any solution to this @mymikemiller

**Comment by darren-f5s** [comment]
Having this issue now when playing videos.

**Comment by cbenhagen** [comment]
@ramiechaarani & @darren-f5s please check solutions posted in https://github.com/flutter/flutter/issues/47524

**Comment by darren-f5s** [comment]
@cbenhagen Thanks for the reply! Unfortunately I can't update to the latest version as I need to keep with flutter build 1.9.x (For older phone compat). Is there anything I can do directly in code to fix this issue my side?

**Comment by darren-f5s** [comment]
@cbenhagen Am I correct in assuming that if I do a local update of the code based off the following changes that it will (should) work?
https://github.com/flutter/plugins/pull/2434/commits/824787e4fdba35691c5285e58c3c0e58666db7bd

**Comment by cbenhagen** [comment]
Yes. But it would be easier to use version 0.10.3+1 of the video player plugin.

**Comment by darren-f5s** [comment]
Agreed! I added the following to my `/ios/Runner/AppDelegate.swift` file (inside `didFinishLaunchingWithOptions`) which fixed the issue - for anyone else interested.

```
    do {
        if #available(iOS 10.0, *){
            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, mode: AVAudioSessionModeDefault)
            try AVAudioSession.sharedInstance().setActive(true)
        }
    } catch {
        print(error)
    }
```

**Comment by cbenhagen** [comment]
That's only a good solution if your app is only playing videos or other sound because it will stop any sound the user may have playing in the background as soon as your app starts.
